### PR TITLE
📦  FEAT: [bulk/main] remove all old signatures on setup

### PR DESCRIPTION
**Pull request by chancesmith**
Screen Shot 2023-01-03 at 11 44 27 AM

### CODE REVIEW

File: csv-bulk-signature/src/main.ts [hunk 1]
```diff
@@ -24,12 +24,28 @@ import { TemplateData, Contact } from "./types";
  * MAIN FUNCTIONS
  */
 async function setupFolders() {
-  // create ./dist/signatures if it doesn't exist
-  if (!fs.existsSync("./dist")) {
-    await fs.promises.mkdir("./dist");
+  await createDirIfMissing("./dist");
+  await createDirIfMissing("./dist/signatures");
+  await deleteOldSignatures();
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:29]: refactor to seperate functions

File: csv-bulk-signature/src/main.ts [hunk 2]
```diff
@@ -24,12 +24,28 @@ import { TemplateData, Contact } from "./types";
...
+}
+
+async function createDirIfMissing(path: string) {
+  if (!fs.existsSync(path)) {
+    await fs.promises.mkdir(path);
   }
-  if (!fs.existsSync("./dist/signatures")) {
-    await fs.promises.mkdir("./dist/signatures");
+}
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:36]: refactor to a function that checks for path and creates if needed

File: csv-bulk-signature/src/main.ts [hunk 3]
```diff
@@ -24,12 +24,28 @@ import { TemplateData, Contact } from "./types";
...
+    let count = 0;
+    for await (const file of signaturesFolder) {
+      await fs.promises.unlink(`${SIGNATURES_PATH}/${file}`);
+
+      count++;
+    }
+
+    console.log(`🗑️  CLEAN UP: Deleted ${count} old signatures`);
   }
 }
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:50]: delete all files in signature path and log out

File: csv-bulk-signature/src/main.ts [hunk 4]
```diff
@@ -68,12 +84,6 @@ async function generateSignatures(
   console.log("👍 Signatures generated");
 }
 
-const checkRequiredFields = (row: Contact) =>
-  !!row["Brand*"].length &&
-  !!row["Full Name*"].length &&
-  !!row["Title*"].length &&
-  !!row["Office Phone*"].length;
-
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:76]: more validation down

File: csv-bulk-signature/src/main.ts [hunk 5]
```diff
@@ -163,12 +173,16 @@ async function main() {
...
+    !!row["Title*"].length &&
+    !!row["Office Phone*"].length;
+
   const contactsWithNewProps = contacts
     .filter(skipFirstPlaceholderRow)
-    .map((contact) => ({
-      ...contact,
-      skip: !checkRequiredFields(contact),
-    }));
+    .map((contact) => ({ ...contact, skip: !checkRequiredFields(contact) }));
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:185]: trimming line fat 🍟

**Review by adamnitti — APPROVED** [review]

**Pull request merged**